driver/src/order_driver.rs

```diff
@@ -36,10 +36,15 @@ pub fn run_order_listener<D, C>(
 
 
             let mut orders = db.get_orders_of_slot(slot.low_u32())?;
-            let mut order_hash = orders.rolling_hash(0);
+            let order_hash = orders.rolling_hash(0);
             hash_consistency_check(order_hash, contract_order_hash, "order")?;
 
             let standing_orders = db.get_standing_orders_of_slot(slot.low_u32())?;
+            let standing_order_index = db.get_standing_orders_index_of_slot(slot.low_u32())?;
+            let order_hash_calculated = calculate_order_hash(order_hash, standing_orders.clone());
+            let order_hash_from_contract = contract.calculate_order_hash(slot, standing_order_index.clone())?;
+            hash_consistency_check(order_hash_calculated, order_hash_from_contract, "order")?;
```

Comment:
I think we should be able to differentiate the two errors if we decide to do a consistency check both for non-reserved as well as for all orders.

So I think we should either wrap the error, or pass a custom string that gets reflected in the error string.

Reply:
good suggestions